### PR TITLE
Cli support compute_unit_price

### LINKED ISSUE #25966: Add option to CLI to set prioritization fee
#### Problem

CLI should be able to set `compute-unit-price` to prioritize transactions it sends. 

#### Proposed Solution
- add new option `--compute-unit-price <MICRO-LAMPORTS>    Set the price in micro-lamports of each transaction compute unit`
- include `compute_budget` ix in transactions if above option presents.

### PR DESCRIPTION
**Pull request by taozhu-chicago**
#### Problem
CLI should be able to set compute-unit-price to prioritize transactions it sends.

#### Summary of Changes
Emulating `memo`, adding `compute-unit-price` arg and trait, apply to commands that generate and send transactions.

Fixes #25966

### CODE REVIEW

**Comment by ashpoolin** [comment]
Hi @taozhu-chicago, I saw your changes--this is great, btw--and was eager to test. For me, the argument `--with-compute-unit-price` isn't getting accepted, at least with `solana transfer` feature. Here's what I am seeing:
image

**Comment by taozhu-chicago** [comment]
as example: 
```
./solana transfer testnet-key.json 1 --allow-unfunded-recipient --with-memo "test" --with-compute-unit-price 1 -u testnet  -k testnet-payer.json

Signature: 3oMWGJf1KZLn9VQFnZuyUkvX3DHDSwD2iWd7KeM7HZPZ6Xz1BW6u4hdBpmBVvJEjT1mDi4ke3ogM66fyLEw4Huea
```
https://explorer.solana.com/tx/3oMWGJf1KZLn9VQFnZuyUkvX3DHDSwD2iWd7KeM7HZPZ6Xz1BW6u4hdBpmBVvJEjT1mDi4ke3ogM66fyLEw4Huea?cluster=testnet

**Comment by ashpoolin** [comment]
I tried this and can confirm it worked for me on localnet and devnet. The feature isn't activated (for lack of a more specific word) yet on mainnet, right? Great work, btw!

**Review by apfitzge — DISMISSED** [review]
Code changes look good.
Only thing that feels missing is adding some tests where the compute unit price is not None. Right now we've added this new optional argument and all the tests just use None.

**Comment by taozhu-chicago** [comment]
> Code changes look good. Only thing that feels missing is adding some tests where the compute unit price is not None. Right now we've added this new optional argument and all the tests just use None.

Yeah, was just pretty much duplicated `memo` instruction. Good point about tests tho.

File: cli/src/wallet.rs [hunk 1]
```diff
@@ -671,6 +676,7 @@ pub fn process_transfer(
     fee_payer: SignerIndex,
     derived_address_seed: Option<String>,
     derived_address_program_id: Option<&Pubkey>,
+    compute_unit_price: &Option<u64>,
```
Comment (CriesofCarrots) [inline cli/src/wallet.rs:679]: I know that we're inconsistent with `&Option<T>` vs `Option<&T>` throughout the cli, but I'd prefer `Option<&u64>` for all these

File: clap-utils/src/compute_unit_price.rs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    name: "compute_unit_price",
+    long: "--with-compute-unit-price",
+    help: "Set the price in micro-lamports of each transaction compute unit",
+};
+
+pub fn compute_unit_price_arg<'a, 'b>() -> Arg<'a, 'b> {
+    Arg::with_name(COMPUTE_UNIT_PRICE_ARG.name)
+        .long(COMPUTE_UNIT_PRICE_ARG.long)
+        .takes_value(true)
+        .value_name("MICRO-LAMPORTS")
```
Comment (CriesofCarrots) [inline clap-utils/src/compute_unit_price.rs:13]: I thought one of the outcomes of https://github.com/solana-labs/solana/pull/26093 was the decision not to expose the term `micro-lamport` to users?
I'm not totally sure what to put here instead; perhaps we need that terminology entry after all...?

File: cli/src/cli.rs [hunk 3]
```diff
@@ -969,13 +989,15 @@ pub fn process_command(config: &CliConfig) -> ProcessResult {
...
             new_authority,
+            compute_unit_price,
         } => process_authorize_nonce_account(
             &rpc_client,
             config,
             nonce_account,
             *nonce_authority,
             memo.as_ref(),
             new_authority,
+            compute_unit_price,
```
Comment (CriesofCarrots) [inline cli/src/cli.rs:1000]:
```suggestion
            compute_unit_price.as_ref(),
```
... if you make the type change I've suggested below.

**Review by CriesofCarrots — COMMENTED** [review]
Generally looks good. One question, one nit.

Also, would it be useful to add an integration test? i don't know if there's a way to configure a TestValidator such that an addition compute-unit fee is required...

File: clap-utils/src/compute_unit_price.rs (hunk 2 above)
Comment (taozhu-chicago) [reply clap-utils/src/compute_unit_price.rs]: Good call! How about just repalce with plain "compute-unit-price", updated -h text as. well

File: cli/src/wallet.rs (hunk 1 above)
Comment (taozhu-chicago) [reply cli/src/wallet.rs]: Sounds good to me, make changes one step at a time

File: clap-utils/src/compute_unit_price.rs [hunk 4]
```diff
@@ -0,0 +1,15 @@
+use {crate::ArgConstant, clap::Arg};
+
+pub const COMPUTE_UNIT_PRICE_ARG: ArgConstant<'static> = ArgConstant {
+    name: "compute_unit_price",
+    long: "--with-compute-unit-price",
+    help: "Set compute unit price to transaction, in increments of 0.000001 lamports per compute unit.",
```
Comment (CriesofCarrots) [inline clap-utils/src/compute_unit_price.rs:6]:
```suggestion
    help: "Set compute unit price for transaction, in increments of 0.000001 lamports per compute unit.",
```

File: clap-utils/src/compute_unit_price.rs (hunk 2 above)
Comment (CriesofCarrots) [reply clap-utils/src/compute_unit_price.rs]: Nice, that works!

**Review by CriesofCarrots — APPROVED** [review]
One nit, then lgtm if CI is happy!

**Comment by taozhu-chicago** [comment]
> Generally looks good. One question, one nit.
> 
> Also, would it be useful to add an integration test? i don't know if there's a way to configure a TestValidator such that an addition compute-unit fee is required...

Not sure about adding integration test here; This PR is to add subcommand, how validator handles compute-unit price perhaps outside this PR scope? Unless you feel strong about it, I'd like to do integrate test separately.

**Comment by CriesofCarrots** [comment]
> how validator handles compute-unit price perhaps outside this PR scope

Yeah, that's fine. I only mentioned that in case it was a thing that TestValidator already supported, in which case it would just be adding an integration test with TestValidator booted with the correct parameters. Maybe later :)
